src/common/cockpitpipe.c

```diff
@@ -405,12 +419,19 @@ dispatch_error (gint fd,
               close_immediately (self, "internal-error");
               return FALSE;
             }
+
+          if (priv->err_forward_to_log)
```

Comment:
Why don't we just check this condition at the start and log directly a `g_warning()` instead of messing around with the buffer?  I almost feel like we could git rid of the separate `err_forward_to_log` by nulling out the `err_buffer` pointer, in fact.

Alternatively, I kinda liked your idea to just close the stderr fd.......

Reply:
I think just continuing to pump stderr is the safest option. Otherwise we might lose important messages.  And with that sudo background watching process being around all the time, we can't actually be sure that nobody writes to stderr even if the bridge has successfully redirected it, no?

Reply:
Right.  We might well see a stderr message from a pam module running at session shutdown, for example...